src/index.ts

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+'use-strict';
+
+import {
+  Contents
+} from 'jupyter-js-services';
+
+import {
+  Widget
+} from 'phosphor-widget';
+
+import './index.css';
+
+/**
+ * A widget which hosts a file browser.
+ */
+export
+class FileBrowser extends Widget {
+
+  /**
+   * Create a new node for the file list.
+   */
+  static createNode(): HTMLElement {
+    var node = document.createElement('div');
+    node.innerHTML = (
+      '<div class="files_inner">' +
+        '<div class="files_header">Files</div>' +
+        '<div class="list_container"></div>' +
+      '</div>'
+    );
+    return node;
+  }
+
+  /**
+   * Construct a new file browser widget.
+   */
+  constructor(baseUrl: string, currentDir: string) {
```

Comment:
We should add optional `contents: IContents` to the constructor to allow for other implementations of `Contents`.